src/Diagnostics/cfl.jl

```diff
@@ -72,31 +73,13 @@ DiffusiveCFL(Δt) = CFL(Δt, cell_diffusion_timescale)
 ##### Accurate CFL via reduction
 #####
 
-using CUDA, CUDAKernels, KernelAbstractions, Tullio
-
 using Oceananigans.Models
-using Oceananigans.Grids: halo_size
-using Oceananigans.Operators: Δxᶠᶜᶜ, Δyᶜᶠᶜ, Δzᶜᶜᶠ
-
-accurate_cell_advection_timescale(model) = accurate_cell_advection_timescale(model.grid, model.velocities)
-
-function accurate_cell_advection_timescale(grid, velocities)
-    Nx, Ny, Nz = size(grid)
-    Hx, Hy, Hz = halo_size(grid)
-
-    is = 1+Hx:Nx+Hx
-    js = 1+Hy:Ny+Hy
-    ks = 1+Hz:Nz+Hz
 
-    u = view(velocities.u.data.parent, is, js, ks)
-    v = view(velocities.v.data.parent, is, js, ks)
-    w = view(velocities.w.data.parent, is, js, ks)
+function accurate_cell_advection_timescale(model)
+    
+    u, v, w = model.velocities 
 
-    min_timescale = minimum(
-        @tullio (min) timescale[k] := 1 / (  abs(u[i, j, k]) / Δxᶠᶜᶜ(i, j, k, grid)
-                                           + abs(v[i, j, k]) / Δyᶜᶠᶜ(i, j, k, grid)
-                                           + abs(w[i, j, k]) / Δzᶜᶜᶠ(i, j, k, grid))
-    )
+    min_timescale = minimum(1 / (abs(u) / Δx + abs(v) / Δy + abs(w) / Δz))
```

Comment:
@simone-silvestri is this right? I think the CFL is actually different in every direction, so it should be something like